### PR TITLE
Log367 pipelines

### PR DESCRIPTION
**Pull request by jcantrill**
This PR resolves https://jira.coreos.com/browse/LOG-367

### CODE REVIEW

File: pkg/apis/logging/v1/pipeline_types.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package v1
+
+//PipelinesSpec specifies log pipelines from a defined sourct to dest endpoings
```
Comment (richm) [inline pkg/apis/logging/v1/pipeline_types.go]: source endpoints

File: pkg/apis/logging/v1/pipeline_types.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+//PipelineTargetSpec specifies destination config for log message endpoints
+type PipelineTargetSpec struct {
+	Type         PipelineTargetType              `json:"type"`
+	Endpoint     string                          `json:"endpoint,omitempty"`
+	Certificates *PipelineTargetCertificatesSpec `json:"certificates,omitempty"`
+}
+
+//PipelineTargetCertificatesSpec specifies secretes for pipelines
```
Comment (richm) [inline pkg/apis/logging/v1/pipeline_types.go]: secrets

File: pkg/apis/logging/v1/pipeline_types.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+type PipelineTargetCertificatesSpec struct {
+	SecreteName string `json:"secretName"`
```
Comment (richm) [inline pkg/apis/logging/v1/pipeline_types.go]: SecretName

File: pkg/generators/pipelines/fluentd/fluent_conf.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+func newSourceLabelCopyConf(source string, targets []logging.PipelineTargetSpec) *sourceLabelCopyConf {
+
+	counters := newTargetTypeCounterMap()
+	targetLabels := []string{}
+	for _, target := range targets {
+		if counter, ok := counters.bump(target.Type); ok {
+			targetLabels = append(targetLabels, reLabelName(source, target.Type, counter))
+		} else {
+			logrus.Warnf("Pipline targets include an unrecognized type: %s", target.Type)
```
Comment (richm) [inline pkg/generators/pipelines/fluentd/fluent_conf.go]: Pipeline

File: pkg/generators/pipelines/fluentd/generators.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	for _, target := range targets {
+		if counter, ok := counters.bump(target.Type); ok {
+			conf := newOutputLabelConf(engine.Template, source, target, counter)
+			result, err := engine.Execute("outputLabelConf", conf)
+			if err != nil {
+				return nil, fmt.Errorf("Error generating fluentd config Processing template %s: %v", engine.Template.Name(), err)
+			}
+			configs = append(configs, result)
+		} else {
+			logrus.Warnf("Pipline targets include an unrecognized type: %s", target.Type)
```
Comment (richm) [inline pkg/generators/pipelines/fluentd/generators.go]: Pipeline

**Comment by jcantrill** [comment]
closing in favor of #231

**Pull request closed without merging**